lib/HLSL/HLUtil.cpp

```diff
@@ -157,10 +157,25 @@ void analyzePointer(const Value *V, PointerStatus &PS, DxilTypeSystem &typeSys,
       for (unsigned i = 0; i < argSize; i++) {
         Value *arg = CI->getArgOperand(i);
         if (V == arg) {
-          // Do not replace struct arg.
-          // Mark stored and loaded to disable replace.
-          PS.MarkAsStored();
-          PS.MarkAsLoaded();
+          if (bLdStOnly) {
+            auto &paramAnnot = annotation->GetParameterAnnotation(i);
```

Comment:
Is it possible for annotation to be nullptr?  I think we should integrate it into condition: `if (bLdStOnly && annotation)`.

Oh, I just checked and see that an above codeblock should continue when annotation is nullptr.  That's not as locally obvious as I wish it was.